design/one-pager-terrajet-metadata-extraction.md

```diff
@@ -0,0 +1,272 @@
+- A concept of metadata extractors from Terraform registry and potentially from
+  other sources for Terrajet-based providers, 
+- A new Terrajet codegen pipeline to generate example manifests, which can
+optionally be invoked in Terrajet-based providers during code generation,
+accepting the scraped metadata from the Terraform registry. 
+- Extension of existing Terrajet codegen pipelines to also generate
+  documentation on `struct`s and fields.
+
+### Goals
+We would like to achieve the following goals with this proposal:
+- The proposed new pipeline(s) or extensions of the existing code generation
+  pipelines must be optional. If, for example, an example generation pipeline is
+  not configured in a Terrajet-based provider repo, or if the already existing
+  code generation pipeline is not configured to also generate documentation,
+  then the behavior of the configured pipelines should not change. Thus,
+  configuration of the new pipelines or enhancement of existing ones with
+  registry-scraped metada should be optional.
+- Like existing Terrajet pipelines, newly added registry metadata based
+  pipelines should be stable, i.e., running them on the same metadata must
+  always produce the same output. Simiarly, any extension of the existing
+  pipelines with registry metadata must preserve their stability.
+- We would like to have means of correcting/adjusting scraped metadata before
+  it's input to the codegen pipelines. This would allow us to make manual
+  corrections/enhancements on the output of a scraper, or even manually craft
+  complete or semi-complete registry metadata documents, if for example the
+  provider is small (in the number of resources it supports), and an automatic
+  scraper is not immediately available. This will also allow us, if needed, to
+  have different scrapers that produce output in the same metadata format. For
+  instance, we may have a relatively complex scraper for extracting metadata
+  from the Terraform registry, and another relatively simple one that just adds
+  example HCL configurations by reading them from their respective
+  [files][aws-example-configurations]. This will allow different scraper
+  implementations to be able to fetch metadata from different sources but the
+  Terrajet pipelines will always be working on a well defined format regardless
+  of how those metadata are scraped.
+- We would like to have the scrapers run as needed, produce their output in the
+  common metadata format, and to have the metadata documents added to their
+  respective repositories. However, we can then have the corresponding pipelines
+  run each time with a `make generate`, just like the existing codegen pipelines
+  we have. This would allow us to separate the lifecycles of metadata-scraping
+  and code generation.
+
+### Metadata Format
+The proposed syntax for scraped metadata documents is YAML as we would also like
+the metadata to be human readable, searchable and maintainable, if needed. A
```

Comment:
What is meant by `maintainable`?

Reply:
I have added example maintenance tasks for the scraped metadata in the document.